### PR TITLE
add: objectName in fav folder

### LINKED ISSUE #8549: Add related object name to the favorite in navigation drawer
## Expected behavior

Image

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=32481-370934&node-type=frame&t=VdD2YzDgfojXH6t0-0

### PR DESCRIPTION
**Pull request by harshrajeevsingh**
Closes: #8549 

It was quite complex to get this right. So, I went through Notion's website to see how they implemented it. 
Instead of using `display: none` or having a space reserved for the Icon, I used clip-path & opacity trick to achieve the desired behaviour. This maintains accessibility and helps in label or ObjectName to take the full space. 

Also, truncation now works for label & objectName as a whole instead of separately, as seen in my previous PR. 

**Caveats**

The only problem that now remains is not having `NavigationDrawerAnimatedCollapseWrapper`. Having it on top of any text or div won't let the flex or truncation property work. 

[Screencast from 2024-11-28 13-37-31.webm](https://github.com/user-attachments/assets/29255cd2-3f15-4b1d-b1e1-c041c70052e5)

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx [hunk 1]
```diff
@@ -163,30 +183,58 @@ const StyledSpacer = styled.span`
...
-  isMobile: boolean;
-  active: boolean;
-}>`
-  margin-left: auto;
-  visibility: ${({ isMobile, active }) =>
-    isMobile || active ? 'visible' : 'hidden'};
+const StyledIcon = styled.div`
+  flex-shrink: 0;
+  flex-grow: 0;
+  margin-right: 8px;
```
Comment (martmull) [inline packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:189]: use theme for `margin-right: ${({theme})} => ...`

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx [hunk 2]
```diff
@@ -233,28 +281,41 @@ export const NavigationDrawerItem = ({
...
-          </NavigationDrawerAnimatedCollapseWrapper>
+          <StyledLabelParent>
+            <StyledEllipsisContainer>
+              <StyledItemLabel>{label}</StyledItemLabel>
+              {objectName ? (
+                <StyledItemObjectName>
+                  {' · '}
+                  {capitalize(objectName)}
+                </StyledItemObjectName>
+              ) : null}
```
Comment (martmull) [inline packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:316]:
```suggestion
              {objectName && (
                <StyledItemObjectName>
                  {' · '}
                  {capitalize(objectName)}
                </StyledItemObjectName>
              )}
```

**Review by martmull — COMMENTED** [review]
Some comments but it looks nice, well done

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx (hunk 1 above)
Comment (harshrajeevsingh) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:189]: Ah! I missed that ;(

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx [hunk 3]
```diff
@@ -233,28 +281,41 @@ export const NavigationDrawerItem = ({
...
+                  !isSettingsPage &&
+                  !isNavigationDrawerExpanded
+                    ? theme.font.color.light
+                    : 'currentColor'
+                }
+              />
+            </StyledIcon>
           )}
 
-          <NavigationDrawerAnimatedCollapseWrapper>
```
Comment (martmull) [inline packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:255]: we need that back

Comment (harshrajeevsingh) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:255]: @martmull I looked into it. But it was not possible. As I described in the description that having `NavigationDrawerAnimatedCollapseWrapper`  on top of any text or div won't let the truncation property work.

**Comment by harshrajeevsingh** [comment]
But, It's working for me 

[Screencast from 2024-11-29 17-42-20.webm](https://github.com/user-attachments/assets/eb1ac4fe-3848-403e-9dd3-c96d3e597c33)

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx (hunk 3 above)
Comment (martmull) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:250]: seen with @Bonapara. We are ok to remove the animated effect

**Review by martmull — APPROVED** [review]

**Pull request merged**